src/glmfit.jl

```diff
@@ -262,16 +262,36 @@ function GeneralizedLinearModel(rr::GlmResp, pp::LinPred,
     return GeneralizedLinearModel(rr, pp, f, fit, 0, NaN, NaN, NaN)
 end
 
-function coeftable(mm::AbstractGLM; level::Real=0.95)
+"""
+    coeftable(model::AbstractGLM; level::Real=0.95, test::Symbol=:z)
+
+For generalized linear models, additional keyword argument `test` allows choosing
+the distribution used for Wald tests of coefficients being different from zero:
+`z` for Normal (the default), `t` for Student.
+"""
+function coeftable(mm::AbstractGLM; level::Real=0.95, test::Symbol=:z)
     cc = coef(mm)
     se = stderror(mm)
     zz = cc ./ se
-    p = 2 * ccdf.(Ref(Normal()), abs.(zz))
-    ci = se * quantile(Normal(), (1 - level) / 2)
+    if test === :z
+        distr = Normal()
+        tstr, pstr = "z", "Pr(>|z|)"
+    elseif test === :t
+        dispersion_parameter(mm.rr.d) ||
+            throw(ArgumentError("`test=:t` is not appropriate for models with " *
+                                "$(nameof(typeof(mm.rr.d))) distribution as " *
+                                "dispersion parameter is not estimated"))
```

Comment:
The support for this claim seems rather weak to me. I could only find https://stat.ethz.ch/pipermail/r-help/2006-March/100878.html. Even in the case of known dispersion, the estimator is only asymptotically normal. I know that it is often claimed that the t/F distribution often has better small-sample properties, but I don't see why that argument shouldn't apply equally in the case of known dispersion parameter (with the exception of the Gaussian case where the small-sample distribution is known to be t/F).

Reply:
Yeah, I was just being cautious because it's not clear why the t distribution would be appropriate here. It seems the only established result is that the ratio asymptotically follows a Normal distribution. The Student distribution only applies to a linear model with the normality assumption. McCullagh & Nelder don't seem to talk about this. I've pushed a commit to allow `test=:t` for all models.